src/components/ContactPreview.tsx

```diff
@@ -0,0 +1,74 @@
+'use client'
+
+import Link from 'next/link'
+import useAvatar from '@/components/useAvatar'
+import { css } from '../../styled-system/css'
+import { useContext } from 'react'
+import { UserContext } from '@/components/UserProvider'
+import { Profile } from '@/types/models'
+import { useContactPreview } from '@/app/domains/contacts/contacts.helpers'
+
+type Props = {
+  contact: Profile
+}
+
+export default function ContactPreview({ contact }: Props) {
+  const { user } = useContext(UserContext)
+
+  const { AvatarPreview } = useAvatar({
+    avatarUrl: contact.avatar_url,
+  })
+
+  const { sharedChats: chats, createChatWithContact } = useContactPreview({
+    user,
+    contact,
+  })
+
+  const handleClick = async () => {
+    if (user?.id && user.id !== contact.id) {
+      createChatWithContact({ userId: user.id, contactId: contact.id })
+    }
+  }
+
+  if (!user) {
+    // TODO: withSession?
+    return null
```

Comment:
cleanup: currently this never happens because this component parent (`messages/new/pages.tsx`) already does the exact same check. Is this correct, or am I missing some other scenario?

Reply:
yes! you are right I can remove that check